app/src/main/java/pl/gittobefit/WorkoutDisplay/adapters/ExerciseListAdapter.java

```diff
@@ -47,18 +67,76 @@ public ExerciseListAdapter(ArrayList<ExerciseExecution> exerciseArrayList) {
 
     @Override
     public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
-        holder.textView1.setText(exerciseArrayList.get(position).getExercise().getName());
+        holder.exerciseName.setText(exerciseArrayList.get(position).getName());
         String text = "";
-        if (exerciseArrayList.get(position).getTime() > 20)
+
+        String properFormSeries = "";
+        switch (exercisesExecutionArrayList.get(position).getSeries())
+        {
+            case 1: properFormSeries = "seria"; break;
+            case 2:
+            case 4:
+            case 3:
+                properFormSeries = "serie"; break;
+            default:  properFormSeries = "serii"; break;
+        }
+
+        String properFormCircuit = "";
+        switch (exercisesExecutionArrayList.get(position).getSeries())
         {
-            text = exerciseArrayList.get(position).getSeries() + " serie, " + exerciseArrayList.get(position).getTime() + " sekund";
+            case 1: properFormCircuit = "obwód"; break;
+            case 2:
+            case 4:
+            case 3:
+                properFormCircuit = "obwody"; break;
+            default:  properFormCircuit = "obwodów"; break;
+        }
+        String properFormRep = "";
+        switch (exercisesExecutionArrayList.get(position).getCount())
+        {
+            case 1: properFormRep = "powtórzenie"; break;
+            case 2:
+            case 4:
+            case 3:
+                properFormRep = "powtórzenia"; break;
+            default:  properFormRep = "powtórzeń"; break;
+        }
+
+        if (exercisesExecutionArrayList.get(position).getTime() > 20)
+        {
+            if (scheduleType.equals("CIRCUIT"))
+            {
+                text = String.format(Locale.getDefault(),"%d %s, %d sekund", exercisesExecutionArrayList.get(position).getSeries(),
+                        properFormCircuit, exercisesExecutionArrayList.get(position).getTime());
```

Comment:
Wydaje mi się, że jeżeli mamy dużo parametrów to lepiej zacząć łamać co parametr